src/edu/washington/escience/myria/io/AmazonS3Source.java

```diff
@@ -0,0 +1,102 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.util.Objects;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.commons.httpclient.URIException;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+/**
+ *
+ */
+@NotThreadSafe
+public class AmazonS3Source implements DataSource, Serializable {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+  /** The logger for debug, trace, etc. messages in this class. */
+  private static final org.slf4j.Logger LOGGER =
+      org.slf4j.LoggerFactory.getLogger(AmazonS3Source.class);
+
+  private final URI s3Uri;
+  private transient ClientConfiguration clientConfig;
+  private transient AmazonS3Client s3Client;
+  private transient GetObjectRequest s3Request;
+
+  private long startRange;
+  private long endRange;
+
+  private final String bucket;
+  private final String key;
+
+  @JsonCreator
+  public AmazonS3Source(@JsonProperty(value = "uri", required = true) final String uri)
```

Comment:
So I guess if only the coordinator ever instantiates query plans with `startRange`/`endRange`, then technically we don't need JSON support for these parameters. I think we still should add encoding support for testing and consistency, though.